### PR TITLE
Processing data models

### PR DESCRIPTION
**Pull request by yemidada**
In this milestone, I will set up the models for or Users, Posts, Comments, and Likes.

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Comment < ApplicationRecord
+  belongs_to :post
```
Comment (nahnahdev) [inline app/models/comment.rb:2]: - [ ] Great job so far @yemidada There are just a few changes I will suggest you implement and you'll be on your way to the next Milestone :wink:
- [ ] From the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png), you can see that a comment belongs to both a user and a post but you have only established the association to a post and not to a user. So kindly fix that :+1:

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Like < ApplicationRecord
+  belongs_to :post
```
Comment (nahnahdev) [inline app/models/like.rb]: - [ ] Just as I mentioned in my earlier comment, from the ERD Diagram you can see that a like also belongs to both a user and a post but you have only established the association to a post and not to a user. So kindly fix that :+1:

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  after_create :update_post_likes_counter
```
Comment (nahnahdev) [inline app/models/like.rb]: - [ OPTIONAL ] Great job making use of one of Rails' active call-back methods. However, I would like to suggest that I will suggest that you consider using another one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) : `after_save` to call the `#update_user_posts_counter` method every time a new post has been saved to database, then make the `update_user_posts_counter` method private. 

- Example Implementation
```ruby
  # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end

```

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @yemidada  , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :fire: 
- Has `User`, `Post`, `Comment`, and `Like` models inside `app/models/`  directory :heavy_check_mark: 
- The associations between models are set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).The associations between models are set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). :heavy_check_mark: 
- User model has a method that returns 3 most recent posts :heavy_check_mark: 
- Post model has a method that updates the posts counter for a user :heavy_check_mark: 
- Post model has a method that returns the 5 most recent comments for a given post :heavy_check_mark: 
- Comment model has a method that updates the comments counter for a post :heavy_check_mark: 
- Like model has a method that updates the likes counter for a post :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by binhussen — APPROVED** [review]
Hi @yemidada ,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Proceed Data Models ✔️
- Well Structured Code ✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**